backend/internal/api/routes_services.go

```diff
@@ -7,14 +7,15 @@ import (
 )
 
 // SetupRoutesWithServices configures all API routes with service access
-func SetupRoutesWithServices(router *gin.Engine, nodeService *service.NodeService, edgeService *service.EdgeService, positionService *service.PositionService, cfg *config.Config) {
+func SetupRoutesWithServices(router *gin.Engine, nodeService *service.NodeService, edgeService *service.EdgeService, positionService *service.PositionService, vaultService service.VaultServiceInterface, cfg *config.Config) {
```

Comment:
TODO(CL): Consider adding nil check for vaultService parameter. Unlike other services, it might be optional during development/testing, but the handlers assume it exists. Either:
1. Make it required and panic if nil
2. Make endpoints return 501 Not Implemented when service is nil
3. Add a feature flag to conditionally register these routes